kie-maven-plugin/src/main/java/org/kie/maven/plugin/KieInMemoryMetaInfoBuilder.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package org.kie.maven.plugin;
+
+import com.google.protobuf.ByteString;
+import org.drools.compiler.builder.impl.KnowledgeBuilderImpl;
+import org.drools.compiler.compiler.PackageRegistry;
+import org.drools.compiler.compiler.io.memory.MemoryFileSystem;
+import org.drools.compiler.kie.builder.impl.InternalKieModule;
+import org.drools.compiler.kie.builder.impl.KieModuleCache;
+import org.drools.compiler.kie.builder.impl.KieModuleCacheHelper;
+import org.drools.core.factmodel.ClassDefinition;
+import org.drools.core.rule.JavaDialectRuntimeData;
+import org.drools.core.rule.KieModuleMetaInfo;
+import org.drools.core.rule.TypeDeclaration;
+import org.drools.core.rule.TypeMetaInfo;
+import org.kie.api.builder.model.KieModuleModel;
+import org.kie.api.definition.KiePackage;
+import org.kie.api.definition.rule.Rule;
+import org.kie.api.definition.type.FactType;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.*;
+/*Copy of KieMetaInfoBuilder but with an in memory FS to expose the KieModuleMetaInfo */
+
+public class KieInMemoryMetaInfoBuilder {
```

Comment:
Is there a way to avoid the copy&paste of the whole class? Could we extract the shared code into a common class?